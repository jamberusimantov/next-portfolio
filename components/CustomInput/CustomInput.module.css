.input {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin: 0 auto;
    padding: 0 1rem;
    position: relative;
}

.input input {
    line-height: 2.5rem;
    font-size: large;
    width: 100%;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.input label {
    position: relative;
    top: .75em;
    line-height: 1.5em;
    width: fit-content;
    padding: 0 .25em;
    margin: 0 .25em;
    font-size: small;
    background-color: white;
}

.label {
    color: Highlight;
}

.error {
    color: red;
}

@media(min-width:768px) {
    .input {
        width: 80%;
    }
}