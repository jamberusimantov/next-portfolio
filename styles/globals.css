html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
    opacity: 1;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.submitBtn {
    width: 100%;
    max-width: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: .5rem;
}

.submitBtn p {
    flex: 1;
    margin: 0;
    font-size: large;
    padding: 0 !important;
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

@media(min-width:580px) {
    .submitBtn {
        margin: auto;
    }
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}