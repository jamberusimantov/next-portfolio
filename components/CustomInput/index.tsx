import React, { forwardRef, ChangeEventHandler, FocusEventHandler, CSSProperties } from "react";
import styles from './CustomInput.module.css';


interface iCustomInputProps {
    title: string;
    value?: string;
    description?: string;
    hasError?: boolean;
    required?: boolean;
    handleBlur?: FocusEventHandler<HTMLInputElement>;
    handleChange?: ChangeEventHandler<HTMLInputElement>;
    type?: keyof { "text": "", "email": "", "tel": "" };
    placeholder?: string;
    style?: CSSProperties;
    testid?: string,
};

const CustomInput = forwardRef<HTMLInputElement, iCustomInputProps>((props, ref) =>
    <div className={styles.input}>
        <label
            htmlFor={props.title}
            className={props.hasError ? styles.error : styles.label}
            children={props.hasError ? " * " + props.description : props.title}
        />
        <input
            ref={ref}
            title={props.title}
            data-testid={props.testid}
            name={props.title}
            value={props.value}
            onChange={props.handleChange}
            onBlur={props.handleBlur}
            type={props.type}
            required={props.required}
            placeholder={props.placeholder}
            style={props.style}
        />
    </div>
);

export default CustomInput;