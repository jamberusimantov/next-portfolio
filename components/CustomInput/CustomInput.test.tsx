/**
 * @jest-environment jsdom
 */
import React from "react";
import { describe, expect, test } from '@jest/globals';
import { render, act } from '@testing-library/react';
import CustomInput from '.';


describe('CustomInput', () => {
    test('use jsdom in this test file', () => {
        expect(document.createElement('div')).not.toBeNull();
    });
    test('renders correctly', async () => {
        let el: HTMLElement = document.createElement("div");
        let getTestId: Function = () => { };
        await act(async () => {
            let { container, getByTestId } = await render(<CustomInput
                title="Language"
                testid="CustomInput"
                value="English"
                handleChange={(e) => { }}
            />);
            el = container;
            getTestId = getByTestId;
        });
        expect(el).toMatchSnapshot();
        expect(getTestId(/CustomInput/).getAttribute("value")).toMatch("English");
        getTestId(/CustomInput/).setAttribute("value", "Hebrew");
        expect(getTestId(/CustomInput/).getAttribute("value")).toMatch("Hebrew");
    });
});