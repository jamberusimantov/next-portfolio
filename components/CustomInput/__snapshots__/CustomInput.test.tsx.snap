// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly 1`] = `
<div>
  <div
    class="input"
  >
    <label
      class="label"
      for="Language"
    >
      Language
    </label>
    <input
      data-testid="CustomInput"
      name="Language"
      title="Language"
      value="English"
    />
  </div>
</div>
`;
